homeassistant/components/atag/__init__.py

```diff
@@ -31,17 +31,34 @@ async def async_setup(hass: HomeAssistant, config):
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up Atag integration from a config entry."""
-    session = async_get_clientsession(hass)
 
-    coordinator = AtagDataUpdateCoordinator(hass, session, entry)
+    async def _async_update_data():
+        """Update data via library."""
+        with async_timeout.timeout(20):
+            try:
+                await atag.update()
+            except AtagException as err:
+                raise UpdateFailed(err) from err
+        return atag
+
+    atag = AtagOne(
+        session=async_get_clientsession(hass), **entry.data, device=entry.unique_id
```

Comment:
For a future PR: using `**entry.data` is dangerous. If you ever would add another piece of information to `entry.data` in a future version, it will prevent users from rolling back.

Reply:
Figured that, will fix next time. Thanks for the quick review!